courses/views/v1/__init__.py

```diff
@@ -190,3 +204,31 @@ def destroy(self, request, *args, **kwargs):
         if deactivated_enrollment is None:
             return Response(status=status.HTTP_400_BAD_REQUEST)
         return Response(status=status.HTTP_204_NO_CONTENT)
+
+    def partial_update(self, request, *args, **kwargs):
+        enrolloment = self.get_object()
+        receive_emails = request.data.get("receive_emails")
+
+        with transaction.atomic():
+            try:
+                if receive_emails:
+                    response = subscribe_for_course_emails(
+                        request.user, enrolloment.run
+                    )
+                    enrolloment.emails_subscription = True
+                else:
+                    response = unsubscribe_for_course_emails(
+                        request.user, enrolloment.run
+                    )
+                    enrolloment.emails_subscription = False
```

Comment:
as it's already checking whether we have to receive emails or not so it's already being set in the if/else. So it's not needed

Reply:
I'm not sure to which of my points you answered that:
-  If it was point#1: I tried to keep boolean as boolean which in a general sense clearer than a string `on`
- If it was point#2: My suggestion would make it a cleaner and a little less code line instead of writing it in two places. we'd have one place it will be charged in case of readability + when a change in this logic would be needed it will be in one place.

Reply:
I said we are filtering out based on `on` or '' so on that basis, I preferred that approach. If we have True/False we could have used that way you are suggesting.